sdk/src/main/java/com/supenta/flitchio/sdk/FlitchioController.java

```diff
@@ -285,161 +280,247 @@ public static int getFlitchioManagerVersionCode(Context context)
+     * Activity's onResume().
+     * If you use this controller in a {@link Service}, this method can be called right after
+     * {@link #onCreate(FlitchioStatusListener)}.
+     * <p/>
+     * After calling this method, you are ensured to receive at least one
+     * {@link FlitchioStatusListener#onFlitchioStatusChanged(Status)} callback with the current
+     * status.
+     * <p/>
+     * This is a variant of {@link #onResume(FlitchioEventListener)} that allows you to
+     * define the thread on which you wish to receive the event callbacks.
      *
-     * @param statusListener The status listener.
-     * @param eventListener  The event listener.
-     * @param handler        The handler associated to the thread on which the callbacks will
-     *                       happen.
-     * @since 0.6.0
-     */
-    public void onResume(FlitchioStatusListener statusListener, FlitchioEventListener eventListener, Handler handler) {
-        synchronized (lockListener) {
+     * @param eventListener The event listener.
+     * @param handler       The handler associated to the thread on which the callbacks will
+     *                      happen.
+     * @see FlitchioController#onResume()
+     * @see FlitchioController#onResume(FlitchioEventListener)
+     * @since 0.7.0
+     */
+    @MainThread
+    public void onResume(FlitchioEventListener eventListener, Handler handler) {
+        /*
+         * SET UP THE STATUS LISTENER
+         */
+        if (statusListener != null) {
             /*
-             * ENSURE CLEAN STATE = termination of the (previous) listeners' thread
+             * DETERMINE THE CURRENT STATUS (as we may have missed updates during pause)
              */
-            resetListener();
+            Status statusAfterCheck;
+            try {
+                if (flitchioService.isConnected(authToken)) {
+                    statusAfterCheck = new Status(Status.CONNECTED);
+                } else {
+                    statusAfterCheck = new Status(Status.DISCONNECTED);
+                }
+            } catch (Exception e) { // NullPointerException | RemoteException
+                // Status is either UNBOUND, BINDING or BINDING_FAILED: we just fire it again
+                statusAfterCheck = this.currentStatus;
+            }
+            FlitchioLog.v("Resumed the controller: current status is " + statusAfterCheck);
 
-            this.statusListener = statusListener;
-            this.eventListener = eventListener;
+            /*
+             * REPORT the determined current status
+             */
+            reportStatus(statusAfterCheck);
 
-            if (statusListener != null) {
-                context.registerReceiver(statusReceiver, statusReceiver.getIntentFilter());
-            }
+            /*
+             * START LISTENING to status updates again
+             */
+            statusReceiver.start(context, new FlitchioStatusListener() {
```

Comment:
Yes I know it's weird cause it's a "trick". The thing is that the broadcast receiver will serve the controller with a status (CONNECTED or DISCONNECTED) that it gets from the broadcast. So the interface between broadcast receiver and controller is the same as between controller and SDK user. I didn't want to recreate a separate interface, but I'm happy to if it clears the confusion!